x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/indexlifecycle/TimeValueScheduleTests.java

```diff
@@ -22,7 +26,15 @@ private TimeValue createRandomTimeValue() {
         return new TimeValue(randomLongBetween(1, 10000), randomFrom(TimeUnit.SECONDS, TimeUnit.MINUTES, TimeUnit.HOURS, TimeUnit.DAYS));
     }
 
-    public void testHascodeAndEquals() {
+    @Before
+    public void setUpStartAndInterval() {
+        // start with random epoch between 1/1/1970 and 31/12/2035 so that start is not
+        // so large such that (start + interval) > Long.MAX
+        start = randomLongBetween(0, 2082672000000L);
```

Comment:
Maybe I'm missing something but I don't quite get the math here - this number + 10,000 days is way less than Long.MAX (2^63)? Are you just giving it plenty of headroom?

Reply:
yeah, why not be more restrictive :). doing randomNonnegativeLong was too large of a range

Reply:
thought I would leave a comment to explain why the magic numbers

Reply:
Alright, cool - I just read this the first time as saying it *had* to be 31/12/2035 (or less) to prevent an overflow, rather than that being somewhat arbitrary.